### PR TITLE
Use single-level directory for e2e test artifacts

### PR DESCRIPTION
**Pull request by pzread**
Instead of having multi-level directories in e2e test artifacts, this change reduces the levels, which makes it easier to batch fetch selected artifacts.

**Background:** `gcloud storage cp` can fetch selected files much faster with a file list, but all files will be saved under the same destination directory. Therefore with multi-level directories, it requires organizing those files back to the multi-level structures after downloading.

A simple solution to this is using a single-level structure, so all files can be downloaded into one directory. The downside is that having all the files in a directory makes it less user-friendly to find the files. But seeing the example below, it looks okay if we encode readable components into file names.

It still has a directory for each IREE module, so we can store the VMFB module and other related flag files together. Those files most likely will be fetched together.

An example of file paths
```
e2e_test_artifacts/model_ebe7897f-5613-435b-a330-3cb967704e5e_EfficientNetV2STF/
e2e_test_artifacts/iree_ebe7897f-5613-435b-a330-3cb967704e5e_EfficientNetV2STF.mlir
e2e_test_artifacts/iree_ebe7897f-5613-435b-a330-3cb967704e5e_EfficientNetV2STF_09cb5300-7f73-45cf-9f68-e114c77ca030/
```

This change also simplifies the code of python modules under `e2e_test_artifacts/*`. We no longer need to build the `ArtifactsRoot` objects, but only provide several helper functions to generate artifact paths.

### CODE REVIEW

File: build_tools/python/e2e_test_artifacts/cmake_generator/iree_rule_generator.py [hunk 1]
```diff
@@ -188,30 +189,53 @@ def _build_target_path(self, target_name: str):
...
+  """
 
   rule_builder = IreeRuleBuilder(package_name=package_name)
 
-  cmake_rules = []
-  for model_dir in artifacts_root.model_dir_map.values():
-    imported_model_artifact = model_dir.imported_model_artifact
-    imported_model = imported_model_artifact.imported_model
-    model_rule = model_rule_map[imported_model.model.id]
+  all_imported_models = collections.OrderedDict(
```
Comment (GMNGeoffrey) [inline build_tools/python/e2e_test_artifacts/cmake_generator/iree_rule_generator.py:209]: As of Python 3.7, regular dictionaries are guaranteed to be ordered, so I don't think you need to use OrderedDict any longer

**Review by GMNGeoffrey — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -739,9 +739,9 @@ jobs:
         run: |
           # Not archiving the directory to allow fetching each file as needed
           # separately.
-          gcloud alpha storage cp -r \
-            "${E2E_TEST_ARTIFACTS_DIR}/iree" \
-            "${E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR}/iree"
+          find "${E2E_TEST_ARTIFACTS_DIR}" -maxdepth 1 -name "iree_*" | \
```
[the remark this answers is not in the document]
Comment (pzread) [reply .github/workflows/ci.yml:745]: The problem is in some cases the IREE benchmarks will need files outside the `iree/` directory. For example, if the model file is linalg MLIR, we don't import it but directly use the downloaded model file. In this case the IREE mlir input file will be under `model/`. So strictly speaking the `find -name "iree_*"` also can't handle such case, but we don't have that case right now. The follow-up change will have a tool to correctly upload all required files.

Added a TODO.

File: build_tools/python/e2e_test_artifacts/cmake_generator/iree_rule_generator.py (hunk 1 above)
Comment (pzread) [reply build_tools/python/e2e_test_artifacts/cmake_generator/iree_rule_generator.py:209]: I was intentionally using OrderedDict to provide a clearer semantic. I'm fine to use dict.
Let's keep it as for consistency and I can have another refactor to remove all OrderedDict usages.

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pzread) [reply .github/workflows/ci.yml]: Ah, I realized we can probably just upload all `model_*.mlir` for now to include MLIR source models. A script to export the dependent MLIR paths can be used once we have the even more complicated cases.

**Pull request merged**